crates/kitsune_p2p/kitsune_p2p/src/spawn/actor.rs

```diff
@@ -264,239 +270,308 @@ impl KitsuneP2pActor {
                         let evt_sender = evt_sender.clone();
                         let host = host.clone();
                         let i_s = i_s.clone();
+
                         async move {
                             let evt_sender = &evt_sender;
+
                             match event {
                                 MetaNetEvt::Connected { remote_url, con } => {
-                                    let _ = i_s.new_con(remote_url, con).await;
+                                    match node_is_authorized(&host, con.peer_id(), Timestamp::now())
+                                        .await
+                                    {
+                                        MetaNetEvtAuth::Authorized => {
+                                            let _ = i_s.new_con(remote_url, con).await;
+                                        }
+                                        MetaNetEvtAuth::UnauthorizedIgnore => {}
+                                        MetaNetEvtAuth::UnauthorizedDisconnect => {
+                                            con.close(
+                                                UNAUTHORIZED_DISCONNECT_CODE,
+                                                UNAUTHORIZED_DISCONNECT_REASON,
+                                            )
+                                            .await;
+                                        }
+                                    }
                                 }
```

Comment:
@maackle as i understand it no, e.g. what would it mean to NOT authorize "disconnect"?

also note that `node_is_authorized` and `nodespace_is_authorized` are totally different

putting it inside the destructure event also makes handling the `con` much easier with the rust type system

Reply:
I don't know what it would mean to not authorize "disconnect". What would it mean to NOT not authorize it?

It just seems cleaner to apply authorization to everything, not only for DRY purposes but because if we ever add another event type, we have to remember to copy this code over.

What about a `match` before this one which just matches on all the cases where we want to authorize (joined with `|`) and then has this code once, and then lists the other cases where authorization is not checked?

It's a suggestion.